kratos/tests/cpp_tests/utilities/test_geometry_utils.cpp

```diff
@@ -45,11 +121,215 @@ namespace Testing {
+        KRATOS_CHECK_NEAR(DN_DX(0,1), 0.0,tolerance);
+        KRATOS_CHECK_NEAR(DN_DX(0,2), 0.0,tolerance);
+        KRATOS_CHECK_NEAR(DN_DX(1,0), 1.0,tolerance);
+        KRATOS_CHECK_NEAR(DN_DX(1,1), -1.0,tolerance);
+        KRATOS_CHECK_NEAR(DN_DX(1,2), 0.0, tolerance);
+        KRATOS_CHECK_NEAR(DN_DX(2,0), 0.0,tolerance);
+        KRATOS_CHECK_NEAR(DN_DX(2,1), 1.0,tolerance);
+        KRATOS_CHECK_NEAR(DN_DX(2,2), -1.0,tolerance);
+        KRATOS_CHECK_NEAR(DN_DX(3,0), 0.0,tolerance);
+        KRATOS_CHECK_NEAR(DN_DX(3,1), 0.0,tolerance);
+        KRATOS_CHECK_NEAR(DN_DX(3,2), 1.0, tolerance);
+    }
+
+    KRATOS_TEST_CASE_IN_SUITE(GeometryUtilsCalculateGeometryDataLine, KratosCoreFastSuite)
+    {
+        Line2D2 <NodeType> line = GenerateExampleLine();
+
+        // Computing the info
+        BoundedMatrix<double,2,1> DN_DX;
+        array_1d<double,2> N;
+        double length;
+
+        GeometryUtils::CalculateGeometryData(line, DN_DX, N, length);
+
+        const double tolerance = 1.0e-8;
+
+        for (int i = 0; i < 2; ++i)
+            KRATOS_CHECK_NEAR(N[i], 1.0/2.0, tolerance);
+
+        KRATOS_CHECK_NEAR(length, 1.0, tolerance);
+
+        KRATOS_CHECK_NEAR(DN_DX(0,0), -1.0,tolerance);
+        KRATOS_CHECK_NEAR(DN_DX(1,0), 1.0,tolerance);
+    }
+
+    KRATOS_TEST_CASE_IN_SUITE(GeometryUtilsSideLenghts2D, KratosCoreFastSuite)
+    {
+        Triangle2D3 <NodeType> triangle = GenerateExampleTriangle();
+
+        // Computing the info
+        double hmin, hmax;
+        GeometryUtils::SideLenghts2D(triangle, hmin, hmax);
+
+        const double tolerance = 1.0e-8;
+
+        KRATOS_CHECK_NEAR(hmin, 1.0,tolerance);
+        KRATOS_CHECK_NEAR(hmax, std::sqrt(2.0),tolerance);
+    }
+
+    KRATOS_TEST_CASE_IN_SUITE(GeometryUtilsCalculateTriangleDistances, KratosCoreFastSuite)
+    {
+        Triangle3D3 <NodeType> triangle = GenerateExampleTriangle3D();
+
+        // Computing the info
+        array_1d<double, 3> distances;
+        distances[0] = 0.1;
+        distances[1] = 0.2;
+        distances[2] = -0.3;
+        GeometryUtils::CalculateTriangleDistances(triangle, distances);
+
+        const double tolerance = 1.0e-6;
+
+        KRATOS_CHECK_NEAR(distances[0], 0.353553, tolerance);
+        KRATOS_CHECK_NEAR(distances[1], 0.392232, tolerance);
+        KRATOS_CHECK_NEAR(distances[2], 0.6, tolerance);
+    }
+
+    KRATOS_TEST_CASE_IN_SUITE(GeometryUtilsCalculateTetrahedraIntersectionPoints, KratosCoreFastSuite)
+    {
+        Tetrahedra3D4 <NodeType> tetrahedra = GenerateExampleTetrahedra();
+
+        // Computing the info
+        array_1d<double, 4> distances;
+        distances[0] = 0.1;
+        distances[1] = 0.2;
+        distances[2] = -0.3;
+        distances[2] = 0.2;
+        array_1d<Point, 4> intersection_points;
+        const int intersections = GeometryUtils::CalculateTetrahedraIntersectionPoints(tetrahedra, distances, intersection_points);
+
+        KRATOS_CHECK_EQUAL(intersections, 1);
```

Comment:
@roigcarlo Is this test
```suggestion
//        KRATOS_CHECK_EQUAL(intersections, 1);
```